docs/config_entries_config_flow_handler.md

```diff
@@ -185,3 +185,92 @@ async def async_migrate_entry(hass, config_entry: ConfigEntry):
 
     return True
 ```
+
+## Reauthentication
+
+Gracefully handling authentication errors such as invalid, expired, or revoked tokens is needed to advance on the [Integration Qualily Scale](integration_quality_scale_index.md). This example of how to add reauth to the OAuth flow created by `script.scaffold` following the pattern in [Building a Python library](api_lib_auth.md#oauth2).
+
+This example catches an authentication exception in config entry setup in `__init__.py` and instructs the user to visit the integrations page in order to reconfigure the integration.
+
+```python
+
+from google_nest_sdm.exceptions import AuthException, GoogleNestException
+from homeassistant.core import HomeAssistant
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Setup up a config entry."""
+
+    # TODO: Replace with actual API setup and exception
+    auth = api.AsyncConfigEntryAuth(...)
+    try:
+        await auth.refresh_tokens()
+    except TokenExpiredError:
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN,
+                context={"source": SOURCE_REAUTH},
+                data=entry.data,
+            )
+        )
+        return False
+
+    # TODO: Proceed with component setup
+```
+
+The flow handler in `config_flow.py` also needs to have some additional steps to support reauth which include showing a confirmation, starting the reauth flow, updating the existing config entry, and reloading to invoke setup again.
+
+```python
+
+class OAuth2FlowHandler(
+    config_entry_oauth2_flow.AbstractOAuth2FlowHandler, domain=DOMAIN
+):
+    """Config flow to handle OAuth2 authentication."""
+
+    async def async_step_reauth(self, user_input=None):
+        """Perform reauth upon an API authentication error."""
+        return await self.async_step_reauth_confirm()
+
+    async def async_step_reauth_confirm(self, user_input=None):
+        """Dialog that informs the user that reauth is required."""
+        if user_input is None:
+            return self.async_show_form(
+                step_id="reauth_confirm",
+                data_schema=vol.Schema({}),
+            )
+        return await self.async_step_user()
+
+    async def async_oauth_create_entry(self, data: dict) -> dict:
+        """Create an oauth config entry or update existing entry for reauth."""
+        entries = self.hass.config_entries.async_entries(DOMAIN)
```

Comment:
If more than one entry is allowed we want to prefer using our unique_id api. I think we should show this instead and/or add a comment about the alternatives here.

The unique_id api could also be used when only one entry is allowed by setting the unique id to the domain eg.

Reply:
I've updated this to use the unique_id api that returns an existing entry, which I think is what you were referring to.  Let me know if this captures what you had in mind.